cdc/puller/multiplexing_puller.go

```diff
@@ -0,0 +1,428 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package puller
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/cdc/kv"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/cdc/processor/tablepb"
+	"github.com/pingcap/tiflow/cdc/puller/frontier"
+	"github.com/pingcap/tiflow/pkg/spanz"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/tikv/client-go/v2/oracle"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	resolveLockFence        time.Duration = 20 * time.Second
+	resolveLockTickInterval time.Duration = 10 * time.Second
+)
+
+type tableProgress struct {
+	changefeed model.ChangeFeedID
+	client     *kv.SharedClient
+	spans      []tablepb.Span
+	startTs    model.Ts
+	tableName  string
+
+	initialized          bool
+	resolvedTsUpdated    time.Time
+	resolvedTs           atomic.Uint64
+	maxIngressResolvedTs atomic.Uint64
+
+	resolvedSpans chan kv.MultiplexingEvent
```

Comment:
```suggestion
	resolvedEventsCache chan kv.MultiplexingEvent
```
Maybe resolvedEventsCache is makes more sense. Also, please add comments to explain the reason for caching resolved events and processing them asynchronously.